### PR TITLE
Port several updates over from Registrar

### PR DESCRIPTION
**Pull request by kdmccormick**
https://openedx.atlassian.net/browse/EDUCATOR-4044
@edx/masters-neem 
@edx/testeng 

Registrar was cut from this repo, but over time we made several general changes that we believe would be beneficial to codify in the IDA cookiecutter:
 * New-style MIDDLEWARE declaration
 * pytest instead of nose
 * Import sorting with isort
 * pip-sync instead of pip install
 * Constrain Django to >=1.11.23
 * Dynamic Makefile helptext generation
 * Separate Makefile rules for piptools, pylint, et al
 * Remove redundant pii_check (validate checks it already)
 * Importing url patterns directly instead of using string include
 * codecov.yml
 * Hide unhelpful depr warnings
 * validation.in = test.txt + quality.txt

Additional changes:
 * Add make upgrade target to root
 * Run make upgrade
 * Remove .rst warning from make test
 * Change openedx.yaml files to reference people
 * Remove useless pylint suppressions
 * Update requirements README section
 * Remove redundant oauth_urlpatterns

Note: I also opened https://github.com/edx/jenkins-job-dsl/pull/769

### CODE REVIEW

File: openedx.yaml [hunk 1]
```diff
@@ -7,4 +7,7 @@ oeps:
   oep-7:  true  # Python 3
   oep-30:       # PII annotation
     applicable: false
-owner: edx/core
+owner: jmbowman
+supporting_teams:
+  - testeng
+
```
Comment (kdmccormick) [inline openedx.yaml:13]: @jmbowman Can you confirm that this change is OK?

Comment (kdmccormick) [reply openedx.yaml:13]: (confirmed through Slack)

File: {{cookiecutter.repo_name}}/.travis.yml [hunk 2]
```diff
@@ -8,11 +8,10 @@ before_install:
     - "export DISPLAY=:99.0"
     - "sh -e /etc/init.d/xvfb start"
 install:
+    - make ci_requirements
     - pip install -U pip wheel codecov
-    - make requirements
 script:
     - make validate_translations
     - make validate
-    - make pii_check
```
Comment (kdmccormick) [inline {{cookiecutter.repo_name}}/.travis.yml:16]: Note for reviewers: PII checking isn't removed; it's just been rolled into `make validate`

File: README.rst [hunk 3]
```diff
@@ -63,7 +63,15 @@ You will be prompted for a few basic details (described below). These will be us
...
 ~~~~~~~~~~~~
 
-The ``requirements`` files do NOT define package versions for many requirements because you should use the latest versions available when initializing a new project. Once you initialize your project run ``pip freeze``, and update the files in the ``requirements`` directory with pinned versions. Failure to do so could cause unexpected results when running tests or deploying.
+The ``requirements`` files packaged may contain out-of-date version pins.
+Once you initialize your project, run ``make upgrade`` to freshen them,
+and then continue to do so regularly going forward.
+Failure to do so could open your IDA to bugs, security vulnerabilities,
+and other issues.
+
+It is recommended to follow the instructions in
```
Comment (iloveagent57) [inline README.rst:73]: 👍

**Review by iloveagent57 — APPROVED** [review]
Looks great, nice work.

File: {{cookiecutter.repo_name}}/Makefile [hunk 4]
```diff
@@ -1,121 +1,135 @@
...
+
+webbrowser.open("file://" + pathname2url(os.path.abspath(sys.argv[1])))
+endef
+export BROWSER_PYSCRIPT
+BROWSER := python -c "$$BROWSER_PYSCRIPT"
+
+# Generates a help message. Borrowed from https://github.com/pydanny/cookiecutter-djangopackage.
+help: ## display this help message
+	@echo "Please use \`make <target>\` where <target> is one of"
+	@perl -nle'print $& if m{^[\.a-zA-Z_-]+:.*?## .*$$}' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m  %-25s\033[0m %s\n", $$1, $$2}'
```
Comment (jansenk) [inline {{cookiecutter.repo_name}}/Makefile:26]: Could you explain what this does?

Comment (jmbowman) [reply {{cookiecutter.repo_name}}/Makefile:26]: This displays a colorized help message of available `make` targets, kind of like this:

```
Please use `make <target>' where <target> is one of
  clean                     archive and delete most git-ignored files
  detect_changed_source_translations check if translation files are up-to-date
  docs                      build the developer documentation for this repository
  ...
```

Ned came up with a slightly cleaner version that doesn't depend on perl, we should probably switch to that in both cookiecutters: https://github.com/edx/edx-platform/blob/master/Makefile#L9

Comment (jansenk) [reply {{cookiecutter.repo_name}}/Makefile:26]: I can't really make out how it's doing that; is it reading this file and getting the targets and descriptions from the comments next to them?

Comment (jmbowman) [reply {{cookiecutter.repo_name}}/Makefile:26]: Essentially.  Ned wrote a blog post about it (both versions): https://nedbatchelder.com/blog/201804/makefile_help_target.html

**Review by iloveagent57 — APPROVED** [review]

**Pull request merged**